### ISSUE TITLE
High CPU load in the main menu

### ISSUE THREAD

**Issue by Jipok** [yvt/openspades#892]
More than in a regular game. 
Because I am developing scripts for the server, most of the time the client is not connected to the server and is in the main menu. The cooler noise is very disturbing. 
Here is what `perf top` shows:
`30.37% openspades [.] spades::draw::GLWaterRenderer::FFTWaveTank<8>::Run`
I do not know what additional information to provide you.

**Comment by Jipok** [comment]
nvtop:
![image](https://user-images.githubusercontent.com/25588359/79896237-a49af180-83f7-11ea-9b53-c8fdc599d4e0.png)

**Comment by yvt** [comment]
This is an intended behavior. The rendering engine is already running at full power when you are in the main menu.

You can instantly join a server [by passing the server address as a command-line argument](https://github.com/yvt/openspades/blob/4fbe5ee5ce2b94d0a157e2fa810f4db309353265/Resources/Unix/Man/openspades.6.in#L26-L28). Perhaps you could use this functionality instead of keeping the main menu open.

**Comment by Jipok** [comment]
An interesting option. But loading the client takes about 7 seconds. I do not want to increase the testing time. Is there a way to just turn off rendering in the menu?

**Comment by yvt** [comment]
You can disable some parts of the 3D rendering in the main menu by replacing the following line:

https://github.com/yvt/openspades/blob/4fbe5ee5ce2b94d0a157e2fa810f4db309353265/Resources/Scripts/Gui/MainScreen/MainScreenUI.as#L71

with this:

```AngelScript
    @renderer.GameMap = null;
```

Do you compile OpenSpades from source? If that's the case, this should be trivial. Otherwise, put the modified source file in [`$XDG_DATA_HOME/openspades/Resources`](https://github.com/yvt/openspades/wiki/Directories#user-resource-directory)`/Scripts/Gui/MainScreen/MainScreenUI.as` and it should take effect.

> That is, without FPS limitation? But why?

Basically, almost everything in OpenSpades is rendered using the same engine as the one used during gameplay. This design allows us to reuse GUI components in multiple places throughout the program without tailoring them to individual use cases (the fewer codepaths we have, the less engineering effort it takes). You might be surprised to hear that the startup screen actually runs on the software renderer that you can choose to use for a gameplay, and if you set `GameMap` in it, it would actually render the map in the background. The point is, the rendering engine simply doesn't care what it's currently used for.

**Comment by Jipok** [comment]
> Otherwise, put the modified source file in $XDG_DATA_HOME/openspades/Resources/Scripts/Gui/MainScreen/MainScreenUI.as and it should take effect.

And I get a black screen that says "Now loading". Judging by the sounds of the menu works and pressing enter confirms this. The situation with the consumption of resources has become better, but it has become impossible to use the client.
